cmd/gardenlet/app/gardenlet.go

```diff
@@ -339,8 +343,19 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 		return nil, err
 	}
 
+	gardenClusterIdentity := &corev1.ConfigMap{}
+	if err := k8sGardenClient.Client().Get(ctx, kutil.Key(metav1.NamespaceSystem, v1beta1constants.ClusterIdentity), gardenClusterIdentity); err != nil {
```

Comment:
I am getting an error when trying to run `gardenlet` caused by this line:

```
panic: the cache is not started, can not read objects

goroutine 1 [running]:
main.main()
	/Users/i500152/go/src/github.com/gardener/gardener/cmd/gardenlet/main.go:57 +0x2ef
exit status 2
make: *** [start-gardenlet] Error 1
```

I wrapped the error below to make sure that the error is produced exactly by this line here. This should be fixed.

Reply:
As we talked, the problem is that I am using the Client() instead of DirectClient(). Didn't catch this because i have not enabled this feature gate.